src/main/java/com/iexec/core/replicate/ReplicatesService.java

```diff
@@ -168,9 +168,9 @@ public int getNbReplicatesContainingStatus(String chainTaskId, ReplicateStatus..
         return addressReplicates.size();
     }
 
-    public int getNbValidContributedWinners(String chainTaskId, String contributionHash) {
+    public int getNbValidContributedWinners(List<Replicate> replicates, String contributionHash) {
```

Comment:
This method has become static.

Reply:
Static methods can't be mocked so I'm not sure whether it should be marked as static.

Reply:
Ok we can keep it non-static if it's more convenient. Could you kindly add a comment to mention that for the future?

Reply:
Sure, see https://github.com/iExecBlockchainComputing/iexec-core/pull/490/commits/d3208756bc211de878afc6f5075ed62f8561ce7f.
By the way, I'm wondering whether we should keep both version of `getNbValidContributedWinners`: one which would retrieve the replicates, the other obtaining it from the parameters. Both version could be added to all `getNbReplicates...`. I guess they all could be potential race-condition issues.

Reply:
Mocking static methods with mockito seems now possible.
See https://www.baeldung.com/mockito-mock-static-methods
The prerequisite is mockito 3.4.0 and spring framework fetches mockito 4.0.0

Reply:
Good to know! So, let's make all these methods static with a non-static version when needed.